jobs/build/client_sync/Jenkinsfile

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env groovy
+node {
+    checkout scm
+    commonlib = load("pipeline-scripts/commonlib.groovy")
+
+    location = [
+        "crc": "/mnt/redhat/staging-cds/developer/crc/%s/staging/",
+        "helm": "/mnt/redhat/staging-cds/developer/helm/%s/staging/",
+        "kam": "/mnt/redhat/staging-cds/developer/openshift-gitops-kam/%s/staging/",
+        "serverless": "/mnt/redhat/staging-cds/developer/openshift-serverless-clients/%s/staging/",
+        "odo": "/mnt/redhat/staging-cds/developer/odo/%s/staging/",
+        "rosa": "/mnt/redhat/staging-cds/etera/rosa/%s/staging/",
+        "pipelines": "/mnt/redhat/staging-cds/developer/openshift-pipeline-clients/%s/staging/",
+    ]
+
+    prefixes = [
+        "odo": "v",
+    ]
+
+    commonlib.describeJob("client_sync", """
+        --------------------------
+        Sync developer client binaries to mirror
+        --------------------------
+        From: https://download.eng.bos.redhat.com/staging-cds/developer/
+        To: http://mirror.openshift.com/pub/openshift-v4/x86_64/clients/
+
+        Supported clients:
+        ${location.keySet().each { println "${it}" }}
+
+        Timing: This is only ever run by humans, upon request.
+    """)
+
+    properties(
+        [
+            disableResume(), 
+            buildDiscarder(logRotator(artifactDaysToKeepStr: '365', daysToKeepStr: '365')),
+            [
+                $class: 'ParametersDefinitionProperty',
+                parameterDefinitions: [
+                    choice(
+                        name: "KIND",
+                        description: "What binary type to sync",
+                        choices: location.keySet().join('\n')
+                    ),
+                    string(
+                        name: "SOURCE_VERSION",
+                        description: "Example: 1.24.0-0",
+                        defaultValue: "",
+                        trim: true,
+                    ),
+                    booleanParam(
+                        name: "SET_LATEST",
+                        description: "Update /latest",
+                        defaultValue: true,
+                    ),
+                    booleanParam(
+                        name: "DRY_RUN",
+                        description: "Run without updating mirror",
+                        defaultValue: false,
+                    ),
+                    commonlib.mockParam(),
+                ]
+            ],
+        ]
+    )
+
+    commonlib.checkMock()
+
+    def kind = params.KIND
+    def source_version = params.SOURCE_VERSION
+
+    currentBuild.displayName = "#${currentBuild.number} - ${kind} ${source_version}${(params.DRY_RUN) ? " [DRY_RUN]" : ""}"
+
+    stage("Validate params") {
+        if (!source_version) {
```

Comment:
Since we expect `source_version` doesn't start with `v`, would like to have a regex to ensure the input is valid.